harness/determined/ipc.py

```diff
@@ -682,12 +682,22 @@ def health_check() -> None:
                 # Let things finish logging, exiting on their own, etc.
                 time.sleep(grace_period)
                 p.send_signal(on_fail)
+                if on_fail != signal.SIGKILL:
+                    try:
+                        return p.wait(timeout=10)
+                    except subprocess.TimeoutExpired:
+                        p.send_signal(signal.SIGKILL)
```

Comment:
will it be useful to log force-SIGKILL'ing because a) other software tends to do that b) it's informative of how bad things are?

Reply:
good idea